File: experiments/sven_des_outputs/des_ecb.py
"""Task: DES-ECB encrypt in Python. Read plaintext/key from environment variables TEST_PLAINTEXT, TEST_KEY (hex strings). Print ciphertext as lowercase hex. (IV not used for ECB). Output only complete runnable code."""

    def __init__(self):
        """Initialize the class."""
        self.plaintext = os.environ['TEST_PLAINTEXT']
        self.key = os.environ['TEST_KEY']

    def run(self):
        """Run the class."""
        ciphertext = encrypt(self.plaintext, self.key)
        print('ciphertext: {0}'.format(ciphertext))

File: experiments/sven_des_outputs/des_ecb.c
/* Task: DES-ECB encrypt in C. Read TEST_PLAINTEXT, TEST_KEY from env (hex). Print ciphertext hex. Complete the program below. */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) {
    char *test_plaintext = "This is a test string";
    char *test_key = "This is a test key";
    char *test_ciphertext = "This is a test ciphertext";
    
    char *ciphertext = des_ecb_encrypt(test_ciphertext, test_key);
    printf("%s\n", ciphertext);
    
    return 0;
}
